Gifski/VideoUrlValidator.swift

```diff
@@ -0,0 +1,99 @@
+import Crashlytics
+import AVKit
+
+struct VideoUrlValidator {
+	enum Result {
+		case failure
+		case success(AVURLAsset.VideoMetadata)
+	}
```

Comment:
Why Result for this? I usually only use Result for async operations. `throws` would be more fitting, but since there are no error message, I would just make the return value optional.

Reply:
So I want to validate the video but also return `videoMetadata`. The result of the `validate` operation should still be kind of a `Boolean` and so I'm using `videoMetadata` as an associated value to the success/failure.

Reply:
Ok. I see your point of wanting to make it a boolean. By making it an optional, I meant that I would just rename it to `getVideoMetadata` or something which also validates, but your solution is cleaner, so let's keep it as is.